### ISSUE TITLE
Creating new document from template fails to open Collabora Online (NotPermittedException is logged)

### ISSUE THREAD

**Issue by kevinkuan1969** [nextcloud/richdocuments#2697]
Describe the bug
We can create any new document with all the templates except the global template added from the Settings-Adminstration-Office-Global Templates. 

To Reproduce
1. Go to File, select + to add new document  
2. Pick the global template from the selection 
3. Click create
4. Loading new document
5. Document loading failed, please try again later.

Expected behavior

Screenshots

![image](https://user-images.githubusercontent.com/80019930/208930193-7a2ba001-8aa9-4ec7-9607-3f4549138287.png)

Client details:
 - OS - MS Windows 11
 - Browser - MS Edge 
 - Version 108.0.1462.54 (Official build) (64-bit)
 - Device: Desktop

Server details

Operating system: Ubuntu 20.14

Web server: 

Database: MySQL 8.0.31

PHP version: 8.1.13

Nextcloud version: 24.0.7

Version of the Collabora Online - Build-in CODE server: 22.5.802

Version of Nextcloud Office: 6.3.2

<details>
<summary>Logs</summary>

Cannot find any error in the log file and will provide as request

**Comment by juliushaertl** [comment]
Any chance you could share the template file you used and the collabora server logs ?

**Comment by kevinkuan1969** [comment]
![image](https://user-images.githubusercontent.com/80019930/208938228-8e74b958-30ec-4524-9c6c-8e3461a8fb8c.png)
Using the Build-in Code and did not install Collabora Online to any server.

Here the sample of template generated from MS Office 2019

**Comment by kevinkuan1969** [comment]
The sample of template to upload to global template in Nextcloud Office
[Global Template - Realestate.odt](https://github.com/nextcloud/richdocuments/files/10278924/Global.Template.-.Realestate.odt)

**Comment by kevinkuan1969** [comment]
![image](https://user-images.githubusercontent.com/80019930/209305105-3bb07e17-b2f7-4da2-a5b4-fb904736d0b8.png)

Instead of using build-in CODE, i have setup up the new Collaboration Online server hosted the same server. All works perfectly except when select the global template. It failed to load but it works after second attempt.
 
No error found in collabora_error.log file but the the error found in nextcloud.log:

{"reqId":"1mC1toFJdUJeYlcjhHKH","level":3,"time":"2022-12-23T08:40:14+00:00","remoteAddr":"127.0.0.1","user":"--","app":"richdocuments","method":"GET",
"url":"/index.php/apps/richdocuments/wopi/template/3140473?access_token=c5tHYr63eUtpG9bLxY342LHkyJgetbVF","message":"Failed to validate WOPI access",
"userAgent":"COOLWSD HTTP Agent 22.05.9.2","version":"24.0.8.2","exception":{"Exception":"OCP\\Files\\NotPermittedException","Message":"","Code":0,
"Trace":[{"file":"/var/www/nextcloud/lib/private/AppFramework/Middleware/MiddlewareDispatcher.php","line":97,"function":"beforeController","class":
"OCA\\Richdocuments\\Middleware\\WOPIMiddleware","type":"->"},{"file":"/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php","line":125,
"function":"beforeController","class":"OC\\AppFramework\\Middleware\\MiddlewareDispatcher","type":"->"},{"file":"/var/www/nextcloud/lib/private/AppFramewor>

Thanks in advance for the help.

**Comment by thebearon** [comment]
@juliushaertl I can reproduce this as well, with any eg. `.ott` template in a user template folder. Trying to create a template based on the file fails. I could bisect it in the richdocuments repo on the `stable25` branch, with the following result:
https://github.com/nextcloud/richdocuments/commit/7141577519745176d91525130a78bb2d3ef1ec4b
```
Author:     Julius Härtl <jus@bitgrid.net>
AuthorDate: Tue Nov 29 15:40:07 2022 +0100
Commit:     Julius Härtl <jus@bitgrid.net>
CommitDate: Wed Dec 7 17:25:36 2022 +0100

    Move fileid/token check to the WOPIMiddleware
```

**Comment by juliushaertl** [comment]
Thanks for the bisect, i'll check that in detail on Monday.

**Comment by juliushaertl** [comment]
Fix is in https://github.com/nextcloud/richdocuments/pull/2740

**Comment by joshtrichards** [comment]
Fixed in #2740

Which has been merged. Closing.
